velox/tool/QueryTraceToolBase.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/tool/QueryTraceToolBase.h"
+#include <folly/json.h>
+#include "velox/exec/trace/QueryTraceTraits.h"
+#include "velox/exec/trace/QueryTraceUtil.h"
+
+DEFINE_bool(usage, false, "Show the usage");
+DEFINE_string(root, "", "Root dir of the query tracing");
+DEFINE_bool(summary, false, "Show the summary of the tracing");
+DEFINE_bool(pretty, false, "Show the summary of the tracing in pretty JSON");
+DEFINE_string(task_id, "", "Specify the target task id");
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::tool {
+
+QueryTraceToolBase::QueryTraceToolBase()
+    : rootDir_(FLAGS_root), taskId_(FLAGS_task_id) {}
+
+void QueryTraceToolBase::printSummary() const {
+  const auto taskIds = trace::getTaksIds(rootDir_);
+  std::ostringstream summary;
+  summary << "\n++++++Query trace summary++++++\n";
+  summary << "Number of tasks: " << taskIds.size() << "\n";
+  summary << "Task ids: " << folly::join(",", taskIds);
+
+  if (taskIds.empty()) {
+    return;
+  }
+
+  const auto taskId = taskId_.empty() ? taskIds.front() : taskId_;
```

Comment:
Let's list all the task info by default? We can have gflags to control the number of tasks info to display in summary, and if task_id is specified, we only print one particular task? Consider to add a short flag to printSummary() which only print out the number of tasks and task id and user can choose from that? Thanks!

Reply:
Should only record query meta of the first task say task 0?